File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;

    // prompts user for valid input
    do
    {
        printf("Pyramid height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);

    // create number lines
    for (int line = 0; line < height; line++)
    {
        // create number spaces
        for (int spaces = height - line; spaces > 1; spaces--)
        {
            printf(" ");
        }
        // create number hashes
        for (int hashes = 0; hashes < line + 2; hashes++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: pset6/mario/mario.py
import cs50

#get input from user
height = cs50.get_int("Pyramid height:")

#validate input
while height < 0 or height > 23:
    if height > 0 and height < 23:
        break
    height = cs50.get_int("Pyramid height: ")

#for every row print spaces, print hashes, print a new line
for i in range(height):
    # print spaces
    for j in range(height - i -1):
        print(" ", end="")
    # print hashes
    for k in range(i + 2):
        print("#", end="")
    # print a new line
    print()